velox/connectors/hive/iceberg/TransformFactory.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/connectors/hive/iceberg/PartitionSpec.h"
+#include "velox/connectors/hive/iceberg/Transform.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+/// Factory for creating Iceberg partition transforms.
+class TransformFactory {
+ public:
+  /// Create transforms for Iceberg partition specification.
+  ///
+  /// Convenience method that creates a transform for each field in the
+  /// partition specification. This is useful when you need to transform
+  /// multiple columns at once for partitioning.
+  ///
+  /// The returned transforms are in the same order as the fields in the
+  /// partition specification. Each transform is created using createTransform()
+  /// and can be applied to the corresponding column in a row vector.
+  ///
+  /// @param spec The partition specification containing the fields to
+  /// transform. Must not be null.
+  /// @param connectorQueryCtx Connector query context for expression evaluation
+  /// for partition transform. Must not be null.
+  /// @return A vector of transforms, one for each partition field in the spec.
+  /// The vector is empty if the spec contains no fields.
+  /// @throws VeloxUserError if any field's transform type is not supported
+  /// or if the field type is not valid for partitioning.
+  static std::vector<TransformPtr> createTransforms(
```

Comment:
Why do we need Transform class at all? Can we not simply create a vector of core::TypedExprPtr?